### PR TITLE
Integrate pigweed tokenizer logging

### PR DESCRIPTION
**Pull request by andrei-menzopol**
Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

#### Problem
Integrate pigweed tokenizer for logging.

#### Change overview
*add pw_tokenizer build changes.
*add build changes in nxp k32w lighting-app.
*update nxp k32w lighting-app README.
*change logging functions to use the tokenizer.
*guard tokenizer changes with define.
*split logs where too many arguments.
*create build test for k32w lighting-app with tokenizer.
*update .wordlist.txt and remove duplicates.
#### Testing
Tested on NXP K32W061 DK6 board with lighting-app example

### CODE REVIEW

**Comment by andrei-menzopol** [comment]
done

**Review by woody-apple — APPROVED** [review]

**Comment by andrei-menzopol** [comment]
All the required tests now pass. Could you please help me with a review?

File: .github/.wordlist.txt [hunk 1]
```diff
@@ -334,6 +334,9 @@ depottools
 deps
 desc
 descheduled
+detokenization
+detokenizer
+Detokenizer
```
Comment (andy31415) [inline .github/.wordlist.txt:339]: nit: file is not case sensitive, you can run it through `sort -f | unique -i`

File: src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp [hunk 2]
```diff
@@ -806,30 +808,38 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetAndLogThread
         iterCopy = iter;
     }
 
+    memset(printBuf, 0x00, TELEM_PRINT_BUFFER_SIZE);
```
[the remark this answers is not in the document]
Comment (andrei-menzopol) [reply src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp:811]: I've noticed that the  [whole function](https://github.com/project-chip/connectedhomeip/blob/32c4e4359caa14e6020f41967041939e39a21af8/src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp#L672) is guarded for CHIP_PROGRESS_LOGGING.
I believe that src/lib/support/StringBuilder.h doesn't interpret the format string so I think snprintf is needed. Correct me if I'm wrong.
I'm not sure what would be the best approach. I could create another array just for this log message in order to stop using memset.
I will get rid of the first memset, let me know what you think

[the remark this answers is not in the document]
Comment (andy31415) [reply src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp]: I did not notice the larger guard, sorry.
stringbuilder indeed can only support ints and strings. We could eventually add more support however that seems indeed not related to this PR.

**Comment by andrei-menzopol** [comment]
As for the failing tests:

- restyled. It seems that I don't have permissions to change .github/workflows/examples-k32w.yaml
- Run misspelled. Fails because of other words not included by me.
- Darwin. I'm not sure. It was passing before and I haven't made any major changes recently.

I'll rebase

**Review by andy31415 — APPROVED** [review]

**Pull request merged**